runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/WatcherTask.java

```diff
@@ -0,0 +1,27 @@
+package io.aklivity.zilla.runtime.engine.internal.registry;
+
+import java.io.Closeable;
+import java.net.URL;
+import java.util.concurrent.Callable;
+import java.util.function.BiFunction;
+
+import io.aklivity.zilla.runtime.engine.config.NamespaceConfig;
+
+
+public abstract class WatcherTask implements Callable<Void>, Closeable
+{
+    protected final BiFunction<URL, String, NamespaceConfig> changeListener;
+
+    protected WatcherTask(
+        BiFunction<URL, String, NamespaceConfig> changeListener)
+    {
+        this.changeListener = changeListener;
+    }
+
+    public abstract NamespaceConfig watch(
+        URL configURL);
+
+    public abstract void doInitialConfiguration(
+        URL configURL) throws Exception;
+
```

Comment:
See earlier comment on `FileWatcherTask` about potentially removing this method.